MetaMask/metamask-extension
# Support for ENS wildcard and offchain resolution

NOTE: This is retry of https://github.com/MetaMask/metamask-extension/pull/14526 as it included some commits I made but for some reason it didn't show my name 

## Explanation

This PR addresses https://github.com/MetaMask/specifications/issues/9

NOTE: Support for contenthash will be separate PR.

## More information

Please read https://medium.com/the-ethereum-name-service/upgrade-ethers-js-to-5-6-1-to-activate-ens-l2-offchain-integration-40ee1a0fdf2a for more deail.

## Screenshots/Screencaps

### Display forward lookup

<img width="468" alt="Screenshot 2022-04-25 at 16 36 05" src="https://user-images.githubusercontent.com/2630/165173770-9f91f858-74fb-4ff4-b6eb-5ff9773719af.png">

### Display reverse record (aka primary name)

<img width="318" alt="Screenshot 2022-04-25 at 16 35 48" src="https://user-images.githubusercontent.com/2630/165173773-bf3a8b6c-f0da-4f24-a3d5-a74883aeacc1.png">


## Manual testing steps

- 1. Select testnet (Mainnet/Ropsten/Rinkeby/Goerli network)
- 2. Lookup 1.offchainexample.eth to lookup ENS name
- 3. Send ETH to 1.offchainexample.eth and check it shows in the transaction log
